### PR TITLE
Signup page

### PR DESCRIPTION
**Pull request by ramagopal80**
Finished with creating Signup and Login Page UI and DB
Capture
Capture1
Capture2

### CODE REVIEW

**Review by Rexzellius — APPROVED** [review]
DB and UI looks good and completed for this week's MVP.

**Review by korina0430 — CHANGES_REQUESTED** [review]
please remove all bin/lock files and add them to the git ignore

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_13" default="false" project-jdk-name="15" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" default="false" project-jdk-name="15" project-jdk-type="JavaSDK">
```
Comment (korina0430) [inline .idea/misc.xml:4]: Sync with the team on this change. I didn't see it in their PR's. This could break the code on other peoples computers

File: HELP.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ For further reference, please consider the following sections:
 * [Spring Web](https://docs.spring.io/spring-boot/docs/2.4.3/reference/htmlsingle/#boot-features-developing-web-applications)
 * [Spring Boot DevTools](https://docs.spring.io/spring-boot/docs/2.4.3/reference/htmlsingle/#using-boot-devtools)
 * [Thymeleaf](https://docs.spring.io/spring-boot/docs/2.4.3/reference/htmlsingle/#boot-features-spring-mvc-template-engines)
-* [Spring Data JPA](https://docs.spring.io/spring-boot/docs/2.4.3/reference/htmlsingle/#boot-features-jpa-and-spring-data)
+* [Spring Data JPA](https://docs.spring.io/spring-boot/docs/2.4.3/reference/htmlsingle/#boot-features-jpa-and-spring-org.launhcode.healthynutrition.data)
```
Comment (korina0430) [inline HELP.md]: was this change intentional?

File: HELP.md [hunk 3]
```diff
@@ -24,8 +24,8 @@ The following guides illustrate how to use some features concretely:
 * [Serving Web Content with Spring MVC](https://spring.io/guides/gs/serving-web-content/)
 * [Building REST services with Spring](https://spring.io/guides/tutorials/bookmarks/)
 * [Handling Form Submission](https://spring.io/guides/gs/handling-form-submission/)
-* [Accessing Data with JPA](https://spring.io/guides/gs/accessing-data-jpa/)
-* [Accessing data with MySQL](https://spring.io/guides/gs/accessing-data-mysql/)
+* [Accessing Data with JPA](https://spring.io/guides/gs/accessing-org.launhcode.healthynutrition.data-jpa/)
+* [Accessing org.launhcode.healthynutrition.data with MySQL](https://spring.io/guides/gs/accessing-org.launhcode.healthynutrition.data-mysql/)
```
Comment (korina0430) [inline HELP.md]: was this change intentional?

File: build.gradle [hunk 4]
```diff
@@ -6,7 +6,7 @@ plugins {
 
 group = 'org.launchcode'
 version = '0.0.1-SNAPSHOT'
-sourceCompatibility = '13'
+sourceCompatibility = '11'
```
Comment (korina0430) [inline build.gradle:9]: again, check with the team on this

File: build/resources/main/templates/login.html [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+/* When the inputs get focus, do something */
+.form-container input[type=text]:focus, .form-container input[type=password]:focus {
+  background-color: #ddd;
+  outline: none;
+}
+
+/* Set a style for the submit/login button */
+.form-container .btn {
```
Comment (korina0430) [inline build/resources/main/templates/login.html]: Move these styles to a css file

File: build/resources/main/templates/login.html [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        <label >Password</label>
+        <input type="password" placeholder="Enter Password" name="user_pass" required>
+
+        <button type="submit" class="btn login">Login</button>
+        <button type="button" class="btn cancel" onclick="closeForm()">Close</button>
+    </form>
+</div>
+
+<script>
+function openForm() {
```
Comment (korina0430) [inline build/resources/main/templates/login.html]: please add an appropriate js file, do not include this in the html file

File: build/resources/main/templates/login.html [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+}
+</style>
+</head>
+<body>
+
+<!--<h2>Popup Form</h2>-->
+<!--<p>Click on the button at the bottom of this page to open the login form.</p>-->
+<!--<p>Note that the button and the form is fixed - they will always be positioned to the bottom of the browser window.</p>-->
+
+<!--<button class="open-button" onclick="openForm()">Open Form</button>-->
```
Comment (korina0430) [inline build/resources/main/templates/login.html]: can you delete the comments?

File: build/resources/main/templates/login.html [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+<!--<div class="form-popup" id="myForm">-->
```
Comment (korina0430) [inline build/resources/main/templates/login.html]: can you delete this?

File: build/resources/main/templates/login.html [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+function closeForm() {
+  document.getElementById("myForm").style.display = "none";
+}
+</script>
+
+</body>
+</html>
+
+<!--<form method="post">-->
```
Comment (korina0430) [inline build/resources/main/templates/login.html:115]: delete

File: build/resources/main/templates/login.html [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+<!--    <div class="form-group"><label> Username (E-Mail)-->
+<!--        <input type="text" name="user_email">-->
+<!--    </label></div>-->
+<!--    <div class="form-group"><label> Password-->
+<!--        <input type="text" name="user_pass">-->
+<!--    </label></div>-->
+<!--    <input type="submit" value="Login">-->
+
+<!--<form method="post" action="login" modelAttribute = "login">-->
+<!--    <table border="0" cellpadding="2" cellspacing="2">-->
```
Comment (korina0430) [inline build/resources/main/templates/login.html]: delete allt his commented out code

File: build/resources/main/templates/signup.html [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+<html lang="en" xmlns:th="http://www.thymeleaf.org/">
+<head>
+    <meta charset="UTF-8">
+<!--    <h2>Signup</h2>-->
+    <title>Signup Page</title>
+    <style>
+body {font-family: Arial, Helvetica, sans-serif;}
+* {box-sizing: border-box;}
+/* Full-width input fields */
+input[type=text], input[type=password] {
```
Comment (korina0430) [inline build/resources/main/templates/signup.html]: css should be in a css file

File: build/resources/main/templates/signup.html [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+        <div class="clearfix">
+            <button type="button" class="cancelbtn">Cancel</button>
+            <button type="submit" class="signupbtn">Sign Up</button>
+        </div>
+    </div>
+</head>
+
+<!--<body>-->
+<!--<form method="post" action="register" modelAttribute = "register">-->
+<!--    <div><label> First Name *</label> <input type="text" name="user_fname"/> </div>-->
```
Comment (korina0430) [inline build/resources/main/templates/signup.html]: delete

File: build/resources/main/templates/signup.html [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+        <input type="text" placeholder="Enter Email" name="user_email" required>
+
+        <label Password></label>
+        <input type="password" placeholder="Enter Password" name="user_pass" required>
+
+<!--        <label Repeat Password></label>-->
+<!--        <input type="password" placeholder="Repeat Password" name="" required>-->
+
+        <div class="clearfix">
+            <button type="button" class="cancelbtn">Cancel</button>
```
Comment (korina0430) [inline build/resources/main/templates/signup.html]: lets use camel casing in the class names cancelbtn -> cancelBtn

File: build/resources/main/templates/signup.html [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+            <button type="submit" class="signupbtn">Sign Up</button>
```
Comment (korina0430) [inline build/resources/main/templates/signup.html]: signupbtn -> signupBtn

File: src/main/java/org/launhcode/healthynutrition/controllers/HomeController.java [hunk 15]
```diff
@@ -9,7 +9,7 @@
 
     @GetMapping
     @ResponseBody
-    public String index(){
-        return  "Welcome to Healthy Nutrition website!!!!";
+    public String index() {
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/HomeController.java]: nice :)

File: src/main/java/org/launhcode/healthynutrition/controllers/UserController.java [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+import org.launhcode.healthynutrition.models.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/UserController.java]: remove this space please

File: src/main/java/org/launhcode/healthynutrition/controllers/UserController.java [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+public class UserController {
+@Autowired
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/UserController.java]: need a space here

File: src/main/java/org/launhcode/healthynutrition/controllers/UserController.java [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+    public String getLogin() {
+        return "login";
+    }
+
+    @PostMapping("/login")
+    public String displayLoginPage(Model model, User user) {
+        model.addAttribute("user");
+        userRepo.save(user);
+        return "redirect:";
+//        return "/loginPage";
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/UserController.java]: remove comment

File: src/main/java/org/launhcode/healthynutrition/models/User.java [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+    @NotNull
+    private String user_fname;
+    private String user_lname;
+    @Id
+    @Email
+    private String user_email;
+    @NotNull
+    private String user_pass;
+    private String user_repass;
+//    private int id;
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/models/User.java]: remove

File: src/main/java/org/launhcode/healthynutrition/models/User.java [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+@Entity
+public class User {
+
+    @NotNull
+    private String user_fname;
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/models/User.java]: add a space between each variable in this class

File: src/main/java/org/launhcode/healthynutrition/models/User.java [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+
+    public User(){}
+
+    public User(String user_fname, String user_lname, String user_email, String user_pass, int id) {
+        this.user_fname = user_fname;
+        this.user_lname = user_lname;
+        this.user_email = user_email;
+        this.user_pass = user_pass;
+        this.user_repass = user_repass;
+//        this.id = id;
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/models/User.java]: remove

File: src/main/java/org/launhcode/healthynutrition/models/User.java (hunk 19 above)
Comment (korina0430) [reply src/main/java/org/launhcode/healthynutrition/models/User.java]: actually why don't you have an id?

File: src/main/java/org/launhcode/healthynutrition/models/User.java [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+    public int hashCode() {
+        return Objects.hash(getUser_pass(), getUser_repass());
+    }
+
+
+    //    @Override
+//    public String toString() {
+//        return "User [id=" + ", user_fname=" + user_fname + ", user_lname=" + user_lname + ", user_email="
+//                + user_email + ", user_pass=" + user_pass + "]";
+//    }
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/models/User.java]: delete

File: src/main/resources/templates/login.html [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+/* Add some hover effects to buttons */
+.form-container .btn:hover, .open-button:hover {
+  opacity: 1;
+}
+</style>
+</head>
+<body>
+
```
Comment (korina0430) [inline src/main/resources/templates/login.html]: seemsl ike you have these files twice, one in src/ andone in build/ why?

**Review by korina0430 — CHANGES_REQUESTED** [review]

File: .idea/misc.xml (hunk 1 above)
Comment (ramagopal80) [reply .idea/misc.xml:4]: ok changed

File: build/resources/main/templates/login.html (hunk 7 above)
Comment (ramagopal80) [reply build/resources/main/templates/login.html]: done

File: .gitignore [hunk 24]
```diff
@@ -1,5 +1,4 @@
-HELP.md
-.gradle
+#.gradle
```
Comment (korina0430) [inline .gitignore]: This should not be commented out

File: src/main/java/org/launhcode/healthynutrition/controllers/AuthenticationController.java [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
+import java.util.Optional;
+
+@Controller
+public class AuthenticationController {
+    @Autowired
+    UserRepository userRepo;
+
+    private static final String userSessionKey = "user";
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/AuthenticationController.java]: Naming patterns for string constants are USER_SESSION_KEY

File: src/main/java/org/launhcode/healthynutrition/controllers/AuthenticationController.java [hunk 26]
```diff
@@ -0,0 +1,132 @@
...
+            return null;
+        }
+
+        return user.get();
+    }
+
+    private static void setUserInSession(HttpSession session, User user) {
+
+        session.setAttribute(userSessionKey, user.getEmail());
+//       session.setAttribute(userSessionKey, user.getId());
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/controllers/AuthenticationController.java]: please remove all commented code

File: src/main/java/org/launhcode/healthynutrition/models/User.java [hunk 27]
```diff
@@ -0,0 +1,86 @@
...
+
+@Entity
+public class User {
+    @Id
+    @GeneratedValue
+    private Integer Id;
+
+    @NotNull
+    private String firstName;
+    private String lastName;
```
Comment (korina0430) [inline src/main/java/org/launhcode/healthynutrition/models/User.java:19]: minor: add a space between each variable for readability

File: src/main/resources/templates/fragments.html [hunk 28]
```diff
@@ -2,6 +2,26 @@
 <html lang="en" xmlns:th="http://www.thymeleaf.org/">
 <head th:fragment="head">
     <meta charset="UTF-8"/>
+<<<<<<< HEAD
```
Comment (korina0430) [inline src/main/resources/templates/fragments.html]: this is invalid .. please remove comments from the merge conflict

File: src/main/resources/templates/fragments.html [hunk 29]
```diff
@@ -2,6 +2,26 @@
...
+
+<header th:fragment="header">
+    <h1 th:text="${title}">Health & Nutrition</h1>
+    <ul class="nav" style="float:right;">
+        <li class="nav-item"><a class="nav-link" href="/logout">Log out</a></li>
+    </ul>
+
+</header>
+
+=======
```
Comment (korina0430) [inline src/main/resources/templates/fragments.html]: remove

File: src/main/resources/templates/fragments.html [hunk 30]
```diff
@@ -25,5 +45,6 @@ <h2 class="logo">Healthy Nutrition</h2>
         </div>
     </div>
 </nav>
+>>>>>>> main
```
Comment (korina0430) [inline src/main/resources/templates/fragments.html]: remove

**Review by korina0430 — CHANGES_REQUESTED** [review]
there is commented code and invalid html files from having the merge conflict messages

**Comment by ramagopal80** [comment]
image
image
image

File: src/main/resources/static/styles.css [hunk 31]
```diff
@@ -60,8 +67,3 @@ color: red
   .header-left {
     float: none;
   }
-}
```
Comment (korina0430) [inline src/main/resources/static/styles.css:63]: is this curly brace needed?? looks like it is

Comment (ramagopal80) [reply src/main/resources/static/styles.css:63]: yes

**Review by korina0430 — APPROVED** [review]
